src/settings/components/ToyActuatorSettings.tsx

```diff
@@ -0,0 +1,110 @@
+import {
+  ToyActuator,
+  VibrationActuator,
+  ActuatorMode,
+  ActuatorModeLabels,
+} from '../../utils';
+import { PropsWithChildren, useState } from 'react';
+import { SettingsTile } from '../../common';
+import { SettingsDescription } from '../../common/SettingsDescription';
+import { Dropdown } from '../../common/Dropdown';
+import { ActuatorType } from 'buttplug';
+import { Space } from '../../common/Space';
+import { SettingsLabel } from '../../common/SettingsLabel';
+
+export interface ToySettingsProps
+  extends PropsWithChildren<React.HTMLAttributes<HTMLFieldSetElement>> {
+  toyActuator: ToyActuator;
+}
+
+export const ToyActuatorSettings: React.FC<ToySettingsProps> = ({
+  toyActuator,
+}) => {
+  return (
+    <SettingsTile
+      label={`${toyActuator.actuatorType} ${toyActuator.index + 1}`}
+    >
+      {(() => {
+        switch (toyActuator.actuatorType) {
+          case ActuatorType.Vibrate:
+            return <VibratorActuatorSettings toyActuator={toyActuator} />;
+          default:
+            return <UnknownActuatorSettings />;
+        }
+      })()}
+    </SettingsTile>
+  );
+};
+
+export const VibratorActuatorSettings: React.FC<ToySettingsProps> = ({
```

Comment:
This UI is very nested and feels somewhat hard to work with as a result.
I think we should come up with an alternative way to present this. perhaps a dialog would be in order? 

I am still toying with the idea of rewriting all the UI in [MUI](https://mui.com/) but I have limited capacity at the moment. 
It would however make things like this easier as we'd have more access to easy components without reinventing them every spot.

Reply:
I have started doing all that silly UI rewriting buisness. I'll let you know when thats ready so we can maybe think of something for this.

Reply:
Sounds good. UI/UX is not my passion, and the current solution is definitely a bit clunky. I originally wanted to put all of these into drop-down menus, but a dialog box would work too.